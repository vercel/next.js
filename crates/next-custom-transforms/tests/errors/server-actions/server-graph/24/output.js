// exported!
export async function a() {
    // this is not allowed here
    this.foo();
    // arguments is not allowed here
    console.log(arguments);
    const b = async ()=>{
        // this is not allowed here
        this.foo();
        // arguments is not allowed here
        console.log(arguments);
    };
    function c() {
        // this is allowed here
        this.foo();
        // arguments is allowed here
        console.log(arguments);
        const d = ()=>{
            // this is allowed here
            this.foo();
            // arguments is allowed here
            console.log(arguments);
        };
        const e = async ()=>{
            // this is not allowed here
            this.foo();
            // arguments is not allowed here
            console.log(arguments);
        };
    }
}
