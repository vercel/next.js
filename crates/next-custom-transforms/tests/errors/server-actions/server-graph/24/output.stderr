  x "use cache" functions cannot use `this`.
  | 
   ,-[input.js:6:1]
 5 |   // this is not allowed here
 6 |   this.foo()
   :   ^^^^
 7 |   // arguments is not allowed here
   `----
  x "use cache" functions cannot use `arguments`.
  | 
   ,-[input.js:8:1]
 7 |   // arguments is not allowed here
 8 |   console.log(arguments)
   :               ^^^^^^^^^
   `----
  x "use cache" functions cannot use `this`.
  | 
    ,-[input.js:12:1]
 11 |     // this is not allowed here
 12 |     this.foo()
    :     ^^^^
 13 |     // arguments is not allowed here
    `----
  x "use cache" functions cannot use `arguments`.
  | 
    ,-[input.js:14:1]
 13 |     // arguments is not allowed here
 14 |     console.log(arguments)
    :                 ^^^^^^^^^
 15 |   }
    `----
  x Server Actions cannot use `this`.
  | 
    ,-[input.js:33:1]
 32 |       // this is not allowed here
 33 |       this.foo()
    :       ^^^^
 34 |       // arguments is not allowed here
    `----
  x Server Actions cannot use `arguments`.
  | 
    ,-[input.js:35:1]
 34 |       // arguments is not allowed here
 35 |       console.log(arguments)
    :                   ^^^^^^^^^
 36 |     }
    `----
